src/Database/Connection.php

```diff
@@ -278,44 +280,74 @@ public function run(Query $query): StatementInterface
         return $this->getDisconnectRetry()->run(fn () => $this->_driver->run($query));
     }
 
+    /**
+     * Get query factory instance.
+     *
+     * @return \Cake\Database\QueryFactory
+     */
+    public function queryFactory(): QueryFactory
+    {
+        return $this->queryFactory ??= new QueryFactory($this);
+    }
+
     /**
      * Create a new SelectQuery instance for this connection.
      *
+     * @param \Cake\Database\ExpressionInterface|\Closure|array|string|float|int $fields Fields/columns list for the query.
+     * @param array|string $tables List of tables to query.
+     * @param array<string, string> $types Associative array containing the types to be used for casting.
      * @return \Cake\Database\Query\SelectQuery
      */
-    public function newSelectQuery(): SelectQuery
-    {
-        return new SelectQuery($this);
+    public function newSelectQuery(
+        ExpressionInterface|Closure|array|string|float|int $fields = [],
+        array|string $tables = [],
+        array $types = []
+    ): SelectQuery {
```

Comment:
I was initially concerned about all these positional parameters, but users can also used named parameters :smile: Perhaps we should document the named parameter API instead of the positional one.

Reply:
> Perhaps we should document the named parameter API instead of the positional one.

How do you propose we do that? The tools will start complaining if the params aren't documented the "standard" way.

Reply:
> How do you propose we do that?

I was thinking of the examples we provide in the book and in the 'example' blocks in the API docs. I don't think we should change much in how we define types/params in doc blocks.

Reply:
I guess we could do that.